compiler/rustc_hir_pretty/src/lib.rs

```diff
@@ -679,6 +676,10 @@ impl<'a> State<'a> {
     }
 
     fn print_poly_trait_ref(&mut self, t: &hir::PolyTraitRef<'_>) {
+        // FIXME: This isn't correct!
+        if t.modifiers == TraitBoundModifier::Maybe {
+            self.word("?");
+        }
```

Comment:
Question (not-blocking): I'm guessing these are for follow-ups?

Reply:
Yeah, I didn't really want to make any changes to the behavior here. And all of these bugs are on the diagnostic path anyways.